### ISSUE TITLE
1.64.2 shows blank window, side bar, and nothing works

### ISSUE THREAD

**Issue by joneshf** [microsoft/vscode#144031]
Issue Type: <b>Bug</b>

I updated from some version (I think whatever is the lastest of 1.63.x so likely 1.63.2) to 1.64.2 Yesterday. After the update, nothing worked in VS Code. The screen was blank, there were no icons on the side bar, none of the keybindings worked, none of the menu bar items worked. The only thing that worked was closing VS Code.

I downloaded and installed 1.63.2, that version works like normal. I downloaded and installed 1.64.1, that version works like normal. I updated to 1.64.2, and that version is broken the same was as my initial update to 1.64.2.

I'm reporting this bug from version 1.64.1, in case the versions look off or something. It seems like 1.64.2 is reproducibly broken for me. So, lemme know if there's more diagnosis I can do by switching versions.

VS Code version: Code 1.64.1 (d6ee99e4c045a6716e5c653d7da8e9ae6f5a8b03, 2022-02-07T17:26:08.977Z)
OS version: Darwin x64 19.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz (16 x 2300)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.32GB free)|
|Process Argv|-psn_0_532610 --crash-reporter-id 950e0fc6-05f3-4b36-b54a-db855c28a361|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (60)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-zipfs|arc|2.3.0
nix-env-selector|arr|1.0.7
nixpkgs-fmt|B4d|0.0.1
vscode-bazel|Baz|0.4.0
Nix|bbe|1.0.1
better-toml|bun|0.3.2
language-prometheus|caa|0.1.2
source-viz-vscode|cow|0.2.8
vscode-eslint|dba|2.2.2
dhall-lang|dha|0.0.4
vscode-dhall-lsp-server|dha|0.0.4
xstateviz|dks|1.0.3
chmod|dle|1.1.1
vscode-beancount-formatter|don|1.4.2
gitlens|eam|11.7.0
EditorConfig|Edi|0.16.4
elm-ls-vscode|elm|2.4.1
prettier-for-handlebars-vscode|emb|3.0.0
prettier-vscode|esb|9.3.0
vscode-monorepo-workspace|fol|1.3.1
vscode-pull-request-github|Git|0.36.2
go|gol|0.31.1
gc-excelviewer|Gra|4.0.50
sentinel|has|0.0.3
terraform|has|2.19.0
haskell|has|1.8.0
vscode-test-explorer|hbe|2.21.1
plantuml|jeb|2.17.2
language-haskell|jus|3.4.0
magit|kah|0.6.27
vscode-codemetrics|kis|1.24.0
pdm-task-provider|kno|1.0.0
beancount|Len|0.8.0
fish-ide|lun|0.4.0
python|ms-|2022.0.1814523869
vscode-pylance|ms-|2022.2.4
jupyter|ms-|2022.1.1301854968
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.6
test-adapter-converter|ms-|0.1.5
vscode-purty|mva|0.6.0
cuelang|nic|0.0.1
angular-console|nrw|17.14.1
ide-purescript|nwo|0.25.11
language-purescript|nwo|0.2.8
vetur|oct|0.35.0
indent-rainbow|ode|8.2.2
vscode-jest|Ort|4.2.1
vscode-yaml|red|1.4.0
vscode-direnv|Rub|0.0.2
vscode-quick-scope|ska|0.2.0
fish-vscode|sky|0.2.1
vscode-fileutils|sle|3.5.0
rewrap|stk|1.16.3
vscode-make|tec|1.0.2
pdf|tom|1.2.0
latex-support|tor|3.10.0
region-marker|txa|1.2.1
vim|vsc|1.22.2
vscode-proto3|zxh|0.5.5

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30443607
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30429731
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstartedc:30433508
vsclayoutctrc:30437038
vsrem710:30416614
vscop841:30438915
dsvsc009:30440023
pythonvsnew555cf:30442237
vsbas813:30436447
vscscmwlcmc:30438804
vscgsvid2:30438800
helix:30440343
vsnot107cf:30443615

```

</details>

**Comment by gjsjohnmurray** [comment]
Does the problem still happen if you launch with the `--disable-extensions` [CLI option](https://code.visualstudio.com/docs/editor/command-line#_working-with-extensions)?

**Comment by joneshf** [comment]
Sorry, I'm not entirely sure how to do that. I don't have a `code` command that I can use from the terminal. I always start it from spotlight. What would I need to type into a terminal to start it with that flag?

**Comment by gjsjohnmurray** [comment]
I don't use macOS much, but instructions at https://code.visualstudio.com/docs/setup/mac#_alternative-manual-instructions might help you.

**Comment by joneshf** [comment]
Tried a couple more versions:

- VS Code version 1.64.2 universal: shows the blank window, side bar, and nothing works.
- VS Code Insiders version 1.65.0 intel: file is corrupt, so cannot start at all.
- VS Code Insiders version 1.65.0 universal: works like normal.

**Comment by gjsjohnmurray** [comment]
1.65.0 is expected to release later this week.

**Comment by joneshf** [comment]
This is what I get until the window goes blank:

```Console
$ /Applications/Visual\ Studio\ Code.app/Contents/Resources/app/bin/code --verbose
[main 2022-03-03T12:35:27.974Z] [File Watcher (node.js)] Request to start watching: /Users/joneshf/Library/Application Support/Code/User (excludes: <none>),/Users/joneshf/Library/Application Support/Code/User/settings.json (excludes: <none>)
[main 2022-03-03T12:35:27.997Z] Starting VS Code
[main 2022-03-03T12:35:27.997Z] from: /Applications/Visual Studio Code.app/Contents/Resources/app
[main 2022-03-03T12:35:27.998Z] args: {
  _: [],
  diff: false,
  add: false,
  goto: false,
  'new-window': false,
  'reuse-window': false,
  wait: false,
  help: false,
  'list-extensions': false,
  'show-versions': false,
  'pre-release': false,
  version: false,
  verbose: true,
  status: false,
  'prof-startup': false,
  'no-cached-data': false,
  'prof-v8-extensions': false,
  'disable-extensions': false,
  'disable-gpu': false,
  'ms-enable-electron-run-as-node': false,
  telemetry: false,
  debugRenderer: false,
  logExtensionHostCommunication: false,
  'skip-release-notes': false,
  'skip-welcome': false,
  'disable-telemetry': false,
  'disable-updates': false,
  'disable-keytar': false,
  'disable-workspace-trust': false,
  'disable-crash-reporter': false,
  'crash-reporter-id': '950e0fc6-05f3-4b36-b54a-db855c28a361',
  'skip-add-to-recently-opened': false,
  'unity-launch': false,
  'open-url': false,
  'file-write': false,
  'file-chmod': false,
  'driver-verbose': false,
  force: false,
  'do-not-sync': false,
  trace: false,
  'force-user-env': false,
  'force-disable-user-env': false,
  'open-devtools': false,
  __sandbox: false,
  'no-proxy-server': false,
  'no-sandbox': false,
  nolazy: false,
  'force-renderer-accessibility': false,
  'ignore-certificate-errors': false,
  'allow-insecure-localhost': false,
  logsPath: '/Users/joneshf/Library/Application Support/Code/logs/20220303T043527'
}
[main 2022-03-03T12:35:28.000Z] Resolving machine identifier...
[main 2022-03-03T12:35:28.000Z] Resolved machine identifier: 7d65885d66af09b888273a327f216e2fc0696808e24363a090bb3308a29a08a5
[main 2022-03-03T12:35:28.000Z] Main->SharedProcess#connect
[main 2022-03-03T12:35:28.019Z] [File Watcher (node.js)] Started watching: '/Users/joneshf/Library/Application Support/Code/User'
[main 2022-03-03T12:35:28.047Z] [File Watcher (node.js)] Started watching: '/Users/joneshf/Library/Application Support/Code/User/settings.json'
[main 2022-03-03T12:35:28.065Z] StorageMainService: creating global storage
[main 2022-03-03T12:35:28.083Z] lifecycle (main): phase changed (value: 2)
[main 2022-03-03T12:35:28.084Z] windowsManager#open
[main 2022-03-03T12:35:28.085Z] windowsManager#open pathsToOpen [
  {
    backupPath: '/Users/joneshf/Library/Application Support/Code/Backups/1646310868735',
    remoteAuthority: undefined
  }
]
[main 2022-03-03T12:35:28.086Z] windowsManager#doOpenEmpty {
  restore: true,
  remoteAuthority: undefined,
  filesToOpen: undefined,
  forceNewWindow: true
}
[main 2022-03-03T12:35:28.087Z] IPC Object URL: Registered new channel vscode:82844af6-8508-41b3-b05d-7ca4cc0efae8.
[main 2022-03-03T12:35:28.088Z] window#validateWindowState: validating window state on 1 display(s) { mode: 1, x: 0, y: 23, width: 1792, height: 1097 }
[main 2022-03-03T12:35:28.088Z] window#validateWindowState: 1 monitor working area { x: 0, y: 23, width: 1792, height: 1097 }
[main 2022-03-03T12:35:28.088Z] window#ctor: using window state { mode: 1, x: 0, y: 23, width: 1792, height: 1097 }
[main 2022-03-03T12:35:28.177Z] windowsManager#open used window count 1 (workspacesToOpen: 0, foldersToOpen: 0, emptyToRestore: 1, emptyToOpen: 0)
[main 2022-03-03T12:35:28.177Z] lifecycle (main): phase changed (value: 3)
[main 2022-03-03T12:35:28.182Z] update#setState idle
[main 2022-03-03T12:35:28.182Z] update#ctor - manual checks only; automatic updates are disabled by user preference
[main 2022-03-03T12:35:28.183Z] resolveShellEnv(): skipped (VSCODE_CLI is set)
[main 2022-03-03T12:35:28.381Z] ExtensionHostStarterWorker created
[main 2022-03-03T12:35:28.496Z] resolveShellEnv(): skipped (VSCODE_CLI is set)
[7158:0303/043528.814967:INFO:CONSOLE(1)] "[uncaught exception]: TypeError: Cannot read property '0' of undefined", source: vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/bootstrap-window.js (1)
[7158:0303/043528.823645:INFO:CONSOLE(1)] "TypeError: Cannot read property '0' of undefined
    at vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/loader.js:6:6200
    at vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1008:826
    at Function.i._invokeFactory (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/loader.js:5:1626)
    at i.complete (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/loader.js:5:1821)
    at i._onModuleComplete (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/loader.js:6:991)
    at i._resolve (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/loader.js:6:510)
    at i.defineModule (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/loader.js:5:6142)
    at v (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/loader.js:6:1702)
    at define (vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/loader.js:6:3234)
    at vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1008:182
    at vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3039:12984", source: vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/bootstrap-window.js (1)
```

Maybe it's the `TypeError`? Or, maybe that's normal?

**Comment by joneshf** [comment]
Okay, found out something weird: it seems to depend on where the file is located. Here's what I did (a couple times to validate):
1. Download the zip for 1.64.2 to `~/Downloads`.
2. Extract the program to `~/Downloads/Visual Studio Code`.
3. Run the program from `~/Downloads/Visual Studio Code` (either clicking it, or from the terminal).
4. VS Code works like normal.
5. Move the program to `/Applications/Visual Studio Code`.
6. VS Code shows a blank window, side bar, and nothing works.
7. Move the program anywhere else on the disk–including a rename in the same directory, like `/Applications/Visual Studio Code 2`.
8. VS Code works like normal.

I dunno what it is about `/Applications/Visual Studio Code` that makes 1.64.2 have an issue, but maybe that's more information.

**Comment by joneshf** [comment]
Possibly related, I've been having random crashes with this computer (browsers, built-in programs, VS Code, restarts, etc.). After updating to 1.64.2, I initially chalked it up to it being an issue with my specific computer. After downgrading to 1.64.1, I dismissed this idea. But now, I'm thinking it might still be relevant. I dunno if that information helps or not, but figured I'd throw it out there.

**Comment by deepak1556** [comment]
@joneshf does the issue now continue to repro with `1.65.0` stable release ? If so, can you try starting with `code --no-cached-data`

**Comment by joneshf** [comment]
Sorry, missed the previous message. Just updated to 1.65.2, looks like everything in back to working like normal. I can downgrade to 1.65.0 if you'd like me to check that for sure, but my problem is fixed on 1.65.2. Thanks again!
